src/Microsoft.DotNet.Wpf/src/PresentationFramework/Microsoft/Win32/FileDialog.cs

```diff
@@ -1099,41 +481,48 @@ internal virtual bool PromptUserIfAppropriate(string fileName)
             return fileExists;
         }
 
-        /// <summary>
-        ///     Implements the actual call to GetOpenFileName or GetSaveFileName.
-        /// </summary>
-        internal abstract bool RunFileDialog(NativeMethods.OPENFILENAME_I ofn);
-
         #endregion Internal Methods
 
-        //---------------------------------------------------
-        //
-        // Internal Properties
-        //
-        //---------------------------------------------------
-        #region Internal Properties
-        /// <summary>
-        ///  In cases where we need to return an array of strings, we return
-        ///  a clone of the array.  We also need to make sure we return a 
-        ///  string[0] instead of a null if we don't have any filenames.
-        /// </summary>
-        internal string[] FileNamesInternal
+        #region Internal and Protected Methods
+
+        private protected override void PrepareDialog(IFileDialog dialog)
         {
-            get
+            base.PrepareDialog(dialog);
+
+            dialog.SetFileName(CriticalFileName);
```

Comment:
SetFileName is also called in CommonItemDialog. I don't think we need to call it here again.

Reply:
Done in https://github.com/dotnet/wpf/pull/7916/commits/ffe135e61299ae37393d7741dbfae9253b0e11ea